### DISCUSSION TITLE
Dropzone, Linux + mounted drives

### DISCUSSION THREAD

**Discussion by tentwofour** [sulu/sulu#5795] [Q&A] [answered]
I just spent a bit of time searching the Sulu admin build for an (non-existent) issue - the included admin build w/Dropzone will not allow you to drag/drop images from an CIFS-mounted share (from Linux anyway) in the media select (single/multiple) or the media library area. 

Unsure if that's security-related on the OS level, browser level, or within the Dropzone config/build itself - the dropzone doesn't appear at all, no errors in the console, nothing. The dropzone is there and initialized however.

I can, however, click to upload the files in the media library with the OS-level dialog, so I don't think it's a mount issue

Tested on Mint 19 with mounted cifs share (from internal network) on Firefox 85.0 and Chrome 83.0.4103.97.

A simple DX/UX improvement would be to either show the Dropzone in the media selection types, or at least some text saying "You can drag/drop items here", similar to the main media library. I had no idea it was available in the media selection until I found an old issue and a merged PR.

**Comment by alexander-schranz** [comment, marked as answer]
@tentwofour when you want to drag and drop things into the media area. Make sure that you before did choose a "Collection". You can only drop into a collection not in the overview. You see that when the "Upload" Button is active, we have an open issue to also implement a Upload Button in the Media Selection Content Type [here](https://github.com/sulu/sulu/issues/5039). 

EDIT: This is not longer the case. If no collection is choosen and you are on the media overview a collection selection will be shown when dropping a media there.

**Reply by tentwofour** [reply]
Yep, found the issue describing that too; this has to be something with my setup, (OS-level?) Just tried on your demo site, with the same results: http://tentwentyfour.ca/hrm.mp4

First attempt at D&D is from a Nemo/gvfs cifs mount, notta. Second attempt is with the same file, locally stored, works as intended.

Glad to hear about the addition of the button - I'll check out the issue.

**Reply by tentwofour** [reply]
Actually, I just tried again, using a mounted share (via /etc/fstab), and that works as intended - so definitely something with GVFS/Nemo mounted shares. Edge use for sure.

**Reply by niklasnatter** [reply]
Sulu uses the [`react-dropzone`](https://github.com/react-dropzone/react-dropzone) package for this functionality. Can you test if this is a problem of the package? 

I think there should be working examples on the [homepage of the package](https://react-dropzone.js.org/).

**Reply by tentwofour** [reply]
Not an issue with react-dropzone, although Nemo-mounted share does fail on their basic example. fstab-mounted share works. Also works with Windows mapped network drive.

The issue is actually with Nemo, I'm not sure why I didn't find this the other day, but it seems to be a [reported issue on linuxmint/nemo](https://github.com/linuxmint/nemo/issues/1924).

Feel free to close this; I'll comment on the [other issue](https://github.com/sulu/sulu/issues/5039)

**Reply by niklasnatter** [reply]
I think discussions cannot be closed - but I will mark the answer as accepted 🙂  Thanks for the feedback!
